compiler/src/dotty/tools/dotc/interactive/Completion.scala

```diff
@@ -114,11 +114,12 @@ object Completion {
     val completions = path match {
         // Ignore synthetic select from `This` because in code it was `Ident`
         // See example in dotty.tools.languageserver.CompletionTest.syntheticThis
-        case Select(qual @ This(_), _) :: _ if qual.span.isSynthetic => completer.scopeCompletions
-        case Select(qual, _) :: _                                    => completer.selectionCompletions(qual)
-        case Import(expr, _) :: _                                    => completer.directMemberCompletions(expr)
-        case (_: untpd.ImportSelector) :: Import(expr, _) :: _       => completer.directMemberCompletions(expr)
-        case _                                                       => completer.scopeCompletions
+        case Select(qual @ This(_), _) :: _ if qual.span.isSynthetic   => completer.scopeCompletions
+        case Select(qual, _) :: _           if !qual.tpe.hasSimpleKind => Map.empty
+        case Select(qual, _) :: _                                      => completer.selectionCompletions(qual)
```

Comment:
I would recommend swapping these 2 cases to avoid the unnecessary negation:
```suggestion
        case Select(qual, _) :: _ if qual.tpe.hasSimpleKind          => completer.selectionCompletions(qual)
        case Select(qual, _) :: _                                    => Map.empty
```